### ISSUE TITLE
Missing types for shipping based on Session object and checkout.session.completed event

### ISSUE THREAD

**Issue by pitops** [stripe/stripe-node#2007]
### Describe the bug

I am utilizing some data that are not typed by the Stripe.Checkout.Session object that I received from checkout.session.completed event

Had to extend the typings usign the following

```
interface ExtendedStripeCheckoutSession extends Stripe.Checkout.Session {
  shipping_rate: string
  shipping: {
    address: {
      city: string
      country: string
      line1: string
      line2: string
      postal_code: string
      state: string
    }
    name: string
  }
}
```

### To Reproduce

Receive a `checkout.session.completed` event by creating a session like the following

```
const session = await stripe.checkout.sessions.create({
    shipping_address_collection: {
      allowed_countries: ['CY']
    },
    shipping_options: ShippingOptions,
    line_items: lineItems,
    mode: 'payment',
    billing_address_collection: 'required',
    phone_number_collection: {
      enabled: true
    },
    success_url: `${process.env.FRONTEND_STORE_URL}/cart?success=1`,
    cancel_url: `${process.env.FRONTEND_STORE_URL}/cart?cancelled=1`,
    metadata: {
      orderId: order.id
    }
  })
```

### Expected behavior

Expected that `shipping` object types and `shipping_rate` types exist on the Session object.

### OS

12.6

### Node version

19.6.0

### Library version

14.14.0

### API version

2023-10-16

**Comment by remi-stripe** [comment]
@pitops the types in the stripe-node SDK always reflect the latest API version. The parameters and properties you used have been deprecated since 2022 when we released the API version [2022-08-01](https://stripe.com/docs/upgrades#2022-08-01)
> The `shipping_rates` parameter has been removed from [create Checkout Session](https://stripe.com/docs/api/checkout/sessions/create). You can use the `shipping_options` parameter instead.
> On the Checkout Session resource, several shipping properties have changed.
`shipping_rate` has been moved into the new `shipping_cost` hash.
`shipping` has been renamed to `shipping_details`.

So those aren't missing from the SDK, they were just removed 18 months ago. I'd recommend migrating to a more recent API version (the latest really) to get all the correct types in the SDKs!
